torch/distributed/distributed_c10d.py

```diff
@@ -659,6 +659,10 @@ def _store_based_barrier(rank, store, group_name, rendezvous_count, timeout, log
     if worker_count == world_size:
         store.set(last_worker_key, "1")
 
+    # adjust the timeout to be at least 10secs + 1sec per thousand ranks to reduce the odds of timeout
+    # this value was empirically found while scale testing.
+    logging_interval = max(logging_interval, timedelta(seconds=10 + world_size / 1000))
```

Comment:
one question on store's wait logic, also to my self:
we know it's the internal that causes "connection reset by peer" issue of TCPStore - the server quits before other clients finish waiting, but isn't this `wait` implemented in asynchronous fashion? i.e. The server would notify clients before exiting therefore server's exit won't stop clients from being notified and proceeding. If not, shall we change the `wait` logic to avoid this issue because increasing the interval looks a hack.

Reply:
The "connection reset by peer" issue was mostly due to all ranks connecting at the same time to the store and overflowing the connection backlog. We addressed that by introducing a wait.

We reflect that additional wait here to reduce the change of having the first barrier produce lots of wait related warnings. 
Plus allowing for longer waits should reduce the load on the store and let it spend more cycles on productive commands (the add ones in this case)..